ui: remove node summary flickering

vladlos opened the pull request:
before: to avoid display wrong values on metrics page - summary
initial load we was hiding this block completly, but this leads to
flickering when new data was requested

after: change condition for hide smmary only when we have no
data to display (initial load). after that we display values as long as
we do not have new to display. do not hide summary block while
request in flight and data become "not valid".

Resolves: #61729

Release note(ui): fix flickering of summary block on metrics page

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/67293)
<!-- Reviewable:end -->

Santamaura reviewed (commented):
> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/67293)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @elkmaster, @koorosh, and @nathanstilwell)

___
*[pkg/ui/src/redux/nodes.ts, line 407 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/67293#-Me0_U3rE692HZ1_sOJY:-Me0_U3sCrtAiolGo760:bs7pgvl) ([raw file](https://github.com/cockroachdb/cockroach/blob/e59f801654fa8064a759e2ce45c0b7dbafd9821b/pkg/ui/src/redux/nodes.ts#L407)):*
> ```XML
> 
> export function selectNodesSummaryEmpty(state: AdminUIState) {
>   return !!state.cachedData.nodes.data;
> ```

What is the purpose of using double NOT op here, wouldn't state.cachedData.nodes.data be the same thing?


<!-- Sent from Reviewable.io -->

nathanstilwell commented:
___
*[pkg/ui/src/redux/nodes.ts, line 407 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/67293#-Me0_U3rE692HZ1_sOJY:-Me0qYRZ6y-OMRMkvlx3:biv6n6s) ([raw file](https://github.com/cockroachdb/cockroach/blob/e59f801654fa8064a759e2ce45c0b7dbafd9821b/pkg/ui/src/redux/nodes.ts#L407)):*
<details><summary><i>Previously, Santamaura (Alex Santamaura) wrote…</i></summary><blockquote>

What is the purpose of using double NOT op here, wouldn't state.cachedData.nodes.data be the same thing?
</blockquote></details>

The "Double Bang" in JavaScript is a common way to coerce a boolean. It's roughly the same thing as `Boolean(state.cachedData.nodes.data)`. So where `state.cachedData.nodes.data` will be `object` or `undefined`, `!!state.cachedData.nodes.data` will be either `true` or `false`.


<!-- Sent from Reviewable.io -->

nathanstilwell reviewed (approved):
> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/67293)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @koorosh and @Santamaura)



<!-- Sent from Reviewable.io -->

koorosh reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/67293)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @elkmaster and @Santamaura)

*[a discussion](https://reviewable.io/reviews/cockroachdb/cockroach/67293#-Me4pVDyBj_UhGG-5vh_:-Me4pVDyBj_UhGG-5vha:b-2o2fvs) (no related file):*
Would it be appropriate to write some test on this change? To test that if we have some dead node then it is displayed below

---
___
*[pkg/ui/src/redux/nodes.ts, line 406 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/67293#-Me4pGHtBPcJNzKOc2w2:-Me4pGHtBPcJNzKOc2w3:b-hkl8zm) ([raw file](https://github.com/cockroachdb/cockroach/blob/e59f801654fa8064a759e2ce45c0b7dbafd9821b/pkg/ui/src/redux/nodes.ts#L406)):*
> ```XML
> }
> 
> export function selectNodesSummaryEmpty(state: AdminUIState) {
> ```

This is a very specific selector and probably it would be good to move it close to the single place where it's used (`pkg/ui/src/views/cluster/containers/nodeGraphs/summaryBar.tsx`)?


<!-- Sent from Reviewable.io -->

vladlos committed:
ui: remove node summary flickering

before: to avoid display wrong values on metrics page - summary
initial load we was hiding this block completly, but this leads to
flickering when new data was requested

after: change condition for hide smmary only when we have no
data to display (initial load). after that we display values as long as
we do not have new to display. do not hide summary block while
request in flight and data become "not valid".

Resolves: cockroachdb#61729

Release note(ui): fix flickering of summary block on metrics page

vladlos reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/67293)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @nathanstilwell and @Santamaura)

*[a discussion](https://reviewable.io/reviews/cockroachdb/cockroach/67293#-Me4pVDyBj_UhGG-5vh_:-MeUkU812WHJLFTkSI1K:bb7f2mf) (no related file):*
<details><summary><i>Previously, koorosh (Andrii Vorobiov) wrote…</i></summary><blockquote>

Would it be appropriate to write some test on this change? To test that if we have some dead node then it is displayed below
</blockquote></details>

done

---
___
*[pkg/ui/src/redux/nodes.ts, line 406 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/67293#-Me4pGHtBPcJNzKOc2w2:-MeUkW2C-oRedSqiump-:bb7f2mf) ([raw file](https://github.com/cockroachdb/cockroach/blob/e59f801654fa8064a759e2ce45c0b7dbafd9821b/pkg/ui/src/redux/nodes.ts#L406)):*
<details><summary><i>Previously, koorosh (Andrii Vorobiov) wrote…</i></summary><blockquote>

This is a very specific selector and probably it would be good to move it close to the single place where it's used (`pkg/ui/src/views/cluster/containers/nodeGraphs/summaryBar.tsx`)?
</blockquote></details>

done


<!-- Sent from Reviewable.io -->

koorosh reviewed (approved):
> Reviewed 4 of 4 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/67293)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Santamaura)



<!-- Sent from Reviewable.io -->

elkmaster commented:
bors r+
